doc/dev/custom_devices.rst

```diff
@@ -17,91 +17,100 @@ Additionally, all measurements will always return ``true``.
+                                        const std::vector<QubitIdType> &, bool) override {}
+            void MatrixOperation(const std::vector<std::complex<double>> &,
+                                        const std::vector<QubitIdType> &, bool) override{}
+
+            auto Expval(ObsIdType) -> double override { return 0.0; }
+            auto Var(ObsIdType) -> double override { return 0.0; }
+            void State(DataView<std::complex<double>, 1> &) override {}
+            void Probs(DataView<double, 1> &) override {}
+            void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) override {}
+            void Sample(DataView<double, 2> &, size_t) override {}
+            void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) override {}
+            void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) override {}
+
+            void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
+                                    const std::vector<QubitIdType> &, size_t) override {}
+
+            void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) override {}
         };
 
-In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+In addition to implementing the ``QuantumDevice`` class, one must implement an entry point for the
+device library with the name ``<DeviceIdentifier>Factory``, where ``DeviceIdentifier`` is used to
+unique the entry point symbol. As an example, we use the identifier ``CustomDevice``:
 
 .. code-block:: c++
 
     extern "C" Catalyst::Runtime::QuantumDevice*
-    getCustomDevice() { return new CustomDevice(); }
+    CustomDeviceFactory(const std::string &kwargs) {
+        return new CustomDevice(kwargs);
+    }
 
-where ``CustomDevice()`` is a constructor for your custom device.
-``CustomDevice``'s destructor will be called by the runtime.
+The entry point function acts as a factory method for the device class. Note that a plugin library
+may also provide several factory methods in case it packages multiple devices into the same
+library. However, it is important that the device identifier be unique, as best as possible, to
+avoid clashes with other plugins.
 
-.. note::
+Importantly, the ``<DeviceIdentifier>`` string in the entry point function needs to match
+exactly what is supplied to the ``__quantum__rt__device("rtd_name", "<DeviceIdentifier>")``
+runtime instruction in compiled user programs, or what is returned from the ``get_c_interface``
+function when integrating the device into a PennyLane plugin. Please see the "Integration with
+Python devices" section further down for details.
+
+``CustomDevice(kwargs)`` serves as a constructor for your custom device, with ``kwargs``
+as a string of device specifications and options, represented in Python dictionary format.
+This may contain information such as device type and a number of shots: ``{'shots': 1000}``.
```

Comment:
Updated version is much clearer, I think we just need to remove `device type` from since there is no context for what it could mean:
```suggestion
An example could bet the number of shots or circuit executions, encoded as follow the following string: ``"{'shots': 1000}"``.
```
(not sure about the quotes or not show the entire thing is a string)

Reply:
Sure, I think adding quotes makes more sense as this highlights the fact that this is passed to device as a string.

Reply:
Thanks! Addressed [6db724c](https://github.com/PennyLaneAI/catalyst/pull/343/commits/6db724cdca8959e77ba934db8aec6e21c5c1f055) and [ab5c257](https://github.com/PennyLaneAI/catalyst/pull/343/commits/ab5c25771f948f00f3ecd0716188fab57553ec3b) with some other fixes to the guideline.